### DISCUSSION TITLE
Prevent log out when using the Magic Link more than once

### DISCUSSION THREAD

**Discussion by nextglabs** [supabase/supabase#11178] [Questions]
I'm using the Magic Link strategy in some of my projects, and I noticed that when I click the generated link twice, the previously logged-in user is logged out. This could be very frustrating for the user, and I saw that the `loginWithOtp` method starts by clearing out the existing session before trying to log in to the user.

I wonder if there is a way, option, or setting within Supabase to prevent this behavior and let the user be logged in instead. 

I'm using NextJS for these projects.

Thanks.

**Comment by th0rgall** [comment]
Hi, I also noticed this, and thought it would be annoying for the users of my system. They might indeed click an old magic link to open a page where they were already logged in before (via valid tokens in local storage). Logging them out seems counter-intuitive. Is there a situation in which this would make sense?

I worked around this status quo by creating a new redirection target page in my SPA that _doesn't_ load the Supabase JS client (important!), removes the error hash parameters in case of an invalid magic link (to avoid this issue), and then redirects to other pages for further processing. 

```html
<!-- Script on the /magic-link page. This is the redirection target for magic links. →
<script>
  if (typeof window != "undefined") {

    // Example failure URL redirect from Gotrue:
    // - http://localhost:3007/magic-link#error=unauthorized_client&error_code=401&error_description=Email+link+is+invalid+or+has+expired
    // Example success URL redirect from Gotrue
    // - http://localhost:3007/magic-link#access_token=SOME_JWT&expires_at=1700242374&expires_in=3600&refresh_token=<some token>&token_type=bearer&type=magiclink
    // - also sets a access & refresh token cookie   

    // Parse the hash/fragment from the URL as search parameters
    const { error, access_token } = Object.fromEntries(
      new URLSearchParams(location.hash.substring(1)).entries()
    );

    if (access_token && !error) {
    // We clicked a valid link!
    // Loading the success page, which loads the Supabase client, which will pick up on the hash in the URL and log in.
      // Preserve the same tokens in the hash so Supabase can pick them up
      window.location.href = `/success-page${location.hash}`;
    }

    if (error && error === "unauthorized_client") {
    // We clicked an invalid link, however, me might still have an valid session in localStorage that Supabase can pick up on.
    // Redirect to the login page, but *don't include the error tokens* so Supabase doesn't pick up on them, and log out the user.
    // The login page should listen to Supabase `SIGNED_IN` and/or `INITIAL_SESSION` events and handle a successful session restore from localStorage, if one is available, and then automatically log in.
      window.location.href = `/login`;
    }
  }
</script>

**Comment by laygir** [comment]
I'm having this same issue as a complaint from my users. :/
